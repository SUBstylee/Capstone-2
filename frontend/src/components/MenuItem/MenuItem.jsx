import './MenuItem.scss';
import {useNavigate} from 'react-router-dom';

const MenuItem = ({ title, imageUrl, linkUrl, match }) => {
    const navigate=useNavigate();

    return (
        <div className={`MenuItem`} onClick={() => navigate(`/search/${linkUrl}`)}>
            <div className='background-image' style={{
                backgroundImage: `url(${imageUrl})`
            }} />
            <div className="content">
                <h1 className="title">{title.toUpperCase()}</h1>
                <span className="subtitle">SHOP NOW</span>
            </div>
        </div >
    )
};

export default MenuItem;