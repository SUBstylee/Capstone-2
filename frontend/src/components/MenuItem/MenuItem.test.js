import { render } from '@testing-library/react';
import MenuItem from './MenuItem.jsx';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

test("renders MenuItem component without crashing", () => {
    const { getByText } = render(
        <Provider store={store}><BrowserRouter><MenuItem
            title='hats'
            imageUrl='/assets/images/shop-img/hats.png'
            id={1}
            linkUrl='__Hats__'
        /></BrowserRouter></Provider>
    );

    expect(getByText('HATS')).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <Provider store={store}><BrowserRouter><MenuItem
            title='hats'
            imageUrl='/assets/images/shop-img/hats.png'
            id={1}
            linkUrl='__Hats__'
        /></BrowserRouter></Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});