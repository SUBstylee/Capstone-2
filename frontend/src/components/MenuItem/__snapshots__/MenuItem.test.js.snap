// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MenuItem"
  >
    <div
      class="background-image"
      style="background-image: url(/assets/images/shop-img/hats.png);"
    />
    <div
      class="content"
    >
      <h1
        class="title"
      >
        HATS
      </h1>
      <span
        class="subtitle"
      >
        SHOP NOW
      </span>
    </div>
  </div>
</DocumentFragment>
`;
